js/flightlog.js

```diff
@@ -209,27 +209,34 @@ function FlightLog(logData) {
     };
 
     function buildFieldNames() {
-        var
-            i;
+        const disabledFields=that.getSysConfig().fields_disabled_mask;
 
         // Make an independent copy
         fieldNames = parser.frameDefs.I.name.slice(0);
 
         // Add names of slow fields which we'll merge into the main stream
         if (parser.frameDefs.S) {
-            for (i = 0; i < parser.frameDefs.S.name.length; i++) {
+            for (let i = 0; i < parser.frameDefs.S.name.length; i++) {
                 fieldNames.push(parser.frameDefs.S.name[i]);
             }
         }
 
         // Add names for our ADDITIONAL_COMPUTED_FIELDS
-        fieldNames.push("heading[0]", "heading[1]", "heading[2]");
-        fieldNames.push("axisSum[0]", "axisSum[1]", "axisSum[2]");
-        fieldNames.push("rcCommands[0]", "rcCommands[1]", "rcCommands[2]", "rcCommands[3]"); // Custom calculated scaled rccommand
-        fieldNames.push("axisError[0]", "axisError[1]", "axisError[2]"); // Custom calculated error field
+        if (!(disabledFields & 1<<7)) { //if GYRO disabled
```

Comment:
I think this will all become a lot better to read (and not require comments) if you extract the checking into a function, e.g. `if (!fieldIsDisabled(FIELD_GYRO)) {` and use this everywhere.